src/components/Api.js

```diff
@@ -1,3 +1,145 @@
 export default class Api {
-  
-}
\ No newline at end of file
+  constructor() {
+    this.baseURL = "https://around.nomoreparties.co";
+  }
+
+  getUserInfo () {
+    return fetch(`${this.baseURL}/v1/group-12/users/me`, {
+      headers: {
+        authorization: "397bd50b-9f39-4bee-ad9c-11e69aa20ec4"
+      }
+    })
+    .then(response => {
+      if (!response.ok) {
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

This code is repeated in every request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the `reference` for the method rather than the call. 

**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.